packages/relay/src/lib/services/hbarLimitService/index.ts

```diff
@@ -328,12 +374,12 @@ export class HbarLimitService implements IHbarLimitService {
 
   /**
    * Gets the spending plan for the given eth address or ip address.
-   * @param {string} ethAddress - The eth address to get the spending plan for.
+   * @param {string} [ethAddress] - The eth address to get the spending plan for.
    * @param {string} [ipAddress] - The ip address to get the spending plan for.
    * @returns {Promise<IDetailedHbarSpendingPlan | null>} - A promise that resolves with the spending plan or null if none exists.
    * @private
    */
-  private async getSpendingPlan(ethAddress: string, ipAddress?: string): Promise<IDetailedHbarSpendingPlan | null> {
+  private async getSpendingPlan(ethAddress?: string, ipAddress?: string): Promise<IDetailedHbarSpendingPlan | null> {
```

Comment:
If the statement is definitive, then there is a conflict in the implementation of the method itself. If the method is not supposed to be callable without arguments, doesn’t the `return false` statement contradict your assertion? Returning false implies that both arguments, `ethAddress` and `ipAddress`, are invalid, leading to a null return.

Regard to the update, this change altered the validity of the `ethAddress` as it can potentially be read as null in cases where it is used for helper queries that do not necessarily involve `originalCallerAddress`, as explained in the PR's description.

Reply:
Let’s move the logic which is related only to the `totalLimit`/`remainingLimit` and doesn’t require IP/ETH addresses into a separate `shouldLimit` method that accepts no arguments, we can call this method in those situations and the current `shouldLimit` would still be calling it from its implementation, but would extend its logic and require ETH and IP address to be passed to it.

Reply:
This change doesn't affect the implementation of `shouldLimit()`, but rather impacts `addExpense()`. Additionally, introducing another method would result in repetitive code and an additional instance to maintain.

I've pushed a fix that reverts `ethAddress` back to a required parameter for both `getSpendingPlan()` and `createBasicSpendingPlan()`. As a result, in the `executeQuery`, `originalCaller` is turned into an empty string if it's undefined before passing it as a parameter to `addExpense()`. Look into this comment for more details: [5066ec9](https://github.com/hashgraph/hedera-json-rpc-relay/pull/3024/commits/5066ec9ae156228e6d017ba9ed1cab6aa050a102).